### DISCUSSION TITLE
nhmg moudles  Compile Errors ，What causes this?

### DISCUSSION THREAD

**Discussion by q1165600785** [CESR-lab/ucla-roms#38] [General]
make `nhmg`
ls: cannot access '*.opt': No such file or directory
fatal: not a git repository: '/root/ucla-roms-main/.git'
cd /root/ucla-roms-main/NHMG/src; make clean; cd .. ; make
rm -f ../lib/libnhmg.a mg_autotune.o mg_cst.o mg_gather.o mg_grids.o mg_horiz_grids.o mg_intergrids.o mg_mpi_exchange.o mg_mpi.o mg_namelist.o mg_netcdf_out.o mg_projection.o mg_relax.o mg_solvers.o mg_tictoc.o mg_vert_grids.o nhmg_debug.o nhmg.o mg_netcdf_out_true.o *.mod
cd src; make
ifort -c -O3 -fp-model source -L/root/ucla-roms-main/NHMG/lib -lnhmg -L/opt/zyj/lib -lnetcdff -L/opt/zyj/lib -lnetcdf -lnetcdf -lm -L/opt/zyj/lib -lnetcdf  -I/opt/zyj/include -I/opt/zyj/include -I/opt/zyj/include                      mg_cst.f90
ifort: remark #10448: Intel(R) Fortran Compiler Classic (ifort) is now deprecated and will be discontinued late 2024. Intel recommends that customers transition now to using the LLVM-based Intel(R) Fortran Compiler (ifx) for continued Windows* and Linux* support, new language support, new language features, and optimizations. Use '-diag-disable=10448' to disable this message.
ifort: warning #10315: specifying -lm before files may supersede the Intel(R) math library and affect performance
ifort -c -O3 -fp-model source -L/root/ucla-roms-main/NHMG/lib -lnhmg -L/opt/zyj/lib -lnetcdff -L/opt/zyj/lib -lnetcdf -lnetcdf -lm -L/opt/zyj/lib -lnetcdf  -I/opt/zyj/include -I/opt/zyj/include -I/opt/zyj/include                      mg_netcdf_out.f90
ifort: remark #10448: Intel(R) Fortran Compiler Classic (ifort) is now deprecated and will be discontinued late 2024. Intel recommends that customers transition now to using the LLVM-based Intel(R) Fortran Compiler (ifx) for continued Windows* and Linux* support, new language support, new language features, and optimizations. Use '-diag-disable=10448' to disable this message.
ifort: warning #10315: specifying -lm before files may supersede the Intel(R) math library and affect performance
mg_netcdf_out.f90(7): error #7013: This module file was not generated by any release of this compiler.   [NETCDF]
  use netcdf
------^
mg_netcdf_out.f90(146): error #6632: Keyword arguments are invalid without an explicit interface.   [PATH]
         path  = trim(final_netcdf_file_name), & ! NetCDF filename created.
---------^
mg_netcdf_out.f90(147): error #6632: Keyword arguments are invalid without an explicit interface.   [CMODE]
         cmode = NF90_64BIT_OFFSET           , & ! 64_bits record, Large File
---------^
mg_netcdf_out.f90(147): error #6404: This name does not have a type, and must have an explicit type.   [NF90_64BIT_OFFSET]
         cmode = NF90_64BIT_OFFSET           , & ! 64_bits record, Large File
-----------------^
mg_netcdf_out.f90(148): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
         ncid  = nc_id                       )
---------^
mg_netcdf_out.f90(145): error #6404: This name does not have a type, and must have an explicit type.   [NF90_CREATE]
    is_err = nf90_create                     ( &
-------------^
mg_netcdf_out.f90(150): error #6404: This name does not have a type, and must have an explicit type.   [NF90_NOERR]
    if (is_err /= nf90_noerr) then
------------------^
mg_netcdf_out.f90(161): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
         ncid  = nc_id  , &
---------^
mg_netcdf_out.f90(162): error #6632: Keyword arguments are invalid without an explicit interface.   [NAME]
         name  = 'dim1' , &
---------^
mg_netcdf_out.f90(163): error #6632: Keyword arguments are invalid without an explicit interface.   [LEN]
         len   = dim1   , &
---------^
mg_netcdf_out.f90(164): error #6632: Keyword arguments are invalid without an explicit interface.   [DIMID]
         dimid = dim1_id)
---------^
mg_netcdf_out.f90(160): error #6404: This name does not have a type, and must have an explicit type.   [NF90_DEF_DIM]
    is_err=NF90_def_dim ( &
-----------^
mg_netcdf_out.f90(171): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
         ncid  = nc_id  , &
---------^
mg_netcdf_out.f90(172): error #6632: Keyword arguments are invalid without an explicit interface.   [NAME]
         name  = 'dim2' , &
---------^
mg_netcdf_out.f90(173): error #6632: Keyword arguments are invalid without an explicit interface.   [LEN]
         len   = dim2   , &
---------^
mg_netcdf_out.f90(174): error #6632: Keyword arguments are invalid without an explicit interface.   [DIMID]
         dimid = dim2_id)
---------^
mg_netcdf_out.f90(184): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
         ncid   = nc_id              , &
---------^
mg_netcdf_out.f90(185): error #6632: Keyword arguments are invalid without an explicit interface.   [NAME]
         name   = trim(final_vname)  , &
---------^
mg_netcdf_out.f90(186): error #6632: Keyword arguments are invalid without an explicit interface.   [XTYPE]
         xtype  = NF90_DOUBLE        , &
---------^
mg_netcdf_out.f90(186): error #6404: This name does not have a type, and must have an explicit type.   [NF90_DOUBLE]
         xtype  = NF90_DOUBLE        , &
------------------^
mg_netcdf_out.f90(187): error #6632: Keyword arguments are invalid without an explicit interface.   [DIMIDS]
         dimids = (/dim1_id,dim2_id/), &
---------^
mg_netcdf_out.f90(188): error #6632: Keyword arguments are invalid without an explicit interface.   [VARID]
         varid  = var_id             )
---------^
mg_netcdf_out.f90(183): error #6404: This name does not have a type, and must have an explicit type.   [NF90_DEF_VAR]
    is_err=NF90_def_var              ( &
-----------^
mg_netcdf_out.f90(197): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
    is_err =  NF90_enddef(ncid = nc_id)
--------------------------^
mg_netcdf_out.f90(197): error #6404: This name does not have a type, and must have an explicit type.   [NF90_ENDDEF]
    is_err =  NF90_enddef(ncid = nc_id)
--------------^
mg_netcdf_out.f90(207): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
         ncid   = nc_id     , &
---------^
mg_netcdf_out.f90(208): error #6632: Keyword arguments are invalid without an explicit interface.   [VARID]
         varid  = var_id    , &
---------^
mg_netcdf_out.f90(209): error #6632: Keyword arguments are invalid without an explicit interface.   [VALUES]
         values = array_r2D )
---------^
mg_netcdf_out.f90(206): error #6404: This name does not have a type, and must have an explicit type.   [NF90_PUT_VAR]
    is_err = NF90_put_var   ( &
-------------^
mg_netcdf_out.f90(218): error #6632: Keyword arguments are invalid without an explicit interface.   [NCID]
    is_err = nf90_close(ncid = nc_id)
------------------------^
mg_netcdf_out.f90(878): catastrophic error: Too many errors, exiting
compilation aborted for mg_netcdf_out.f90 (code 1)
make[2]: *** [Makefile:25: mg_netcdf_out.o] Error 1
make[1]: *** [Makefile:33: all] Error 2
make: *** [Makefile:56: nhmg] Error 2

**Comment by zsm1-hub** [comment]
it seems like you didn't make netcdf-c and netcdf-fortran install correctlly
